src/libraries/System.IO.FileSystem/tests/Base/SymbolicLinks/BaseSymbolicLinks.FileSystemInfo.cs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    // Contains test methods that can be used for FileInfo and DirectoryInfo.
+    public abstract class BaseSymbolicLinks_FileSystemInfo : BaseSymbolicLinks_FileSystem
+    {
+        // Creates and returns FileSystemInfo instance by calling either the DirectoryInfo or FileInfo constructor and passing the path.
+        protected abstract FileSystemInfo GetFileSystemInfo(string path);
+
+        protected override FileSystemInfo CreateSymbolicLink(string path, string pathToTarget)
+        {
+            FileSystemInfo link = GetFileSystemInfo(path);
+            link.CreateAsSymbolicLink(pathToTarget);
+            return link;
+        }
+
+        protected override FileSystemInfo ResolveLinkTarget(string linkPath, string? expectedLinkTarget, bool returnFinalTarget = false)
```

Comment:
```suggestion
        protected override FileSystemInfo ResolveLinkTarget(string linkPath, bool returnFinalTarget = false);
```
Can we remove this param and all the asserts in this method and keep their implementation exclusive to only call ResolveLinkTarget in each type? I think it's better to have single-responsibility and do all assertions in the test methods.